src/models/Album.js

```diff
@@ -1,61 +1,57 @@
 const mongoose = require('mongoose');
 
-const AlbumSchema = new mongoose.Schema({
-  artistName: {
-    type: String,
-    required:  [true, 'Please provide artist name'],
-    maxlength: [100, 'Artist name can not be more than 100 characters'],
-    trim: true,
-  },
-  albumName: {
-    type: String,
-    required:  [true, 'Please provide album name'],
-    maxlength: [100, 'Album name can not be more than 100 characters'],
-    trim: true,
-  },
-  price: {
-    type: Number,
-    required: [true, 'Please provide album price'],
-    min: 0,
-  },
-  image: {
-    type: String,
-    default: '/uploads/example.jpg',
-  },
-  releaseDate: {
-    type: Date,
-    required: false,
-  },
-  category: {
-    type: String,
-    required: [false, 'Please provide music category'],
-    enum: ['pop', 'rock', 'rap'],
-  },
-  spotifyUrl: {
-    type: String,
-    required: false, //CHANGE TO TRUE LATER!!!!!!
-    validate: {
-      validator: function (v) {
-        return /^https:\/\/open\.spotify\.com\/album\/[a-zA-Z0-9]+$/.test(v); //ensures that the URL starts with https://open.spotify.com/album/ followed by alphanumeric characters.
-      },
-      message: 'Invalid Spotify URL',
+const AlbumSchema = new mongoose.Schema(
+  {
+    artistName: {
+      type: String,
+      required: [true, 'Please provide artist name'],
+      maxlength: [300, 'Artist name can not be more than 100 characters'],
+      trim: true,
+    },
+    albumName: {
+      type: String,
+      required: [true, 'Please provide album name'],
+      maxlength: [300, 'Album name can not be more than 100 characters'],
+      trim: true,
+    },
+    price: {
+      type: Number,
+      required: [false, 'Please provide album price'],
+      default: 29,
+      min: 0,
+    },
+    image: {
+      type: String,
+      default: '/uploads/example.jpg',
+    },
+    releaseDate: {
+      type: Date,
+      required: false,
+    },
+    category: {
+      type: String,
+      required: [false, 'Please provide music category'],
```

Comment:
I think we can leave off the message? Or will it still show the message even though the field is not required?

Reply:
@akosasante deleted it, i was trying to figure out why do we have an array here- did you mean that for using this array and error message we should implement populate method somewhere in the code?

Reply:
```suggestion
      required: false,
```

My suggestion is to just make it the value `false` to indicate it is not required.

The array here on \`required\` is just one of the ways Mongoose allows yo uto define the \`required\` option; it doesn't make the category field an array or anything. Overall, Mongoose allows a couple different types for this field: [https://mongoosejs.com/docs/api/schematype.html#SchemaType.prototype.required()](https://mongoosejs.com/docs/api/schematype.html#SchemaType.prototype.required())

It can be a boolean value (`required: true` or `required: false`). It can be an array, with `required: [true, "error message"]` . It can be just the error message (in which case mongoose assumes you want the field to be required) `required: "error message"`, or it can even be a function if you want the field to only be required in some cases.

Reply:
@akosasante sure, will leave it as 'required: false', i guess we could also delete it as it is false by default

Reply:
Yep, either way works!